### ISSUE TITLE
Conversations publishes wrong PGP public key

### ISSUE THREAD

**Issue by msebald** [siacs/Conversations#2100]
#### General information
- **Version:** 1.14.7
- **Device:** OnePlus One
- **Android Version:** Android 6.0.1 Cyanogenmod 13.1.2
- **Server name:** jabber.hot-chilli.net
- **Server software:** Prosody 0.9
- **Installed server modules:** lots of
- **Conversations source:** PlayStore
#### Steps to reproduce
1. Go into account setup
2. Long press the account where I want to (re)publish the public PGP key
3. Press the option for publishing the public PGP key
4. I see a password dialog where it asks for a password. As name (and unfortunally not the JID or mail address of the key) I see a different name as the one I would expect for this account, as the PGP key which has the same mail address as the JID of the account has a different name as the one displayed. Also when a contact checks my JID in mcabber he sees the wrong PGP key id, so it takes the wrong key. Which is also interesting is the fact that I did not allow any other keys than the one corresponding to this JID for Conversations use in OpenKeychain.
#### Expected result

See above. Also it would be maybe helpful to choose the PGP key for default use for this account and maybe the key which should be used to (re)publish for this account. There is no setting in Conversations or I did not find it. So how is the key chosen? Maybe from the JID - which does not work, see above. Also it would be good to allow more than one key in OpenKeychain for Conversations and still be able to publish the correct PGP key.

**Comment by licaon-kter** [comment]
Logs when doing these operations?

**Comment by msebald** [comment]
Not really much...

```
10-30 20:54:56.006 D/conversations(17722): roi@jabber.hot-chilli.net send ping (action=ui,listeners=true)
10-30 20:54:56.014 D/conversations(17722): app switched into foreground
10-30 20:54:56.141 D/KP2AAF  (4650): event: 2048, package = eu.siacs.conversations
10-30 20:54:56.252 D/KP2AAF  (4650): event: 32, package = eu.siacs.conversations
10-30 20:54:56.293 D/KP2AAF  (4650): event: 2048, package = eu.siacs.conversations
10-30 20:54:56.353 I/cm.log.servpro(25167): [forcestop_listener][D]/  onAccessibilityEvent eu.siacs.conversations.ui.ConversationActivity, mListened = false
10-30 20:54:56.471 D/conversations(17722): AxolotlService (roi@jabber.hot-chilli.net): Received PEP device list update from user@domain.tld, processing...
10-30 20:54:56.619 D/KP2AAF  (4650): event: 2048, package = eu.siacs.conversations
10-30 20:54:57.112 D/KP2AAF  (4650): event: 2048, package = eu.siacs.conversations
10-30 20:54:57.617 D/KP2AAF  (4650): event: 32, package = eu.siacs.conversations
10-30 20:54:59.260 I/ActivityManager(926): START u0 {cmp=eu.siacs.conversations/.ui.ManageAccountActivity} from uid 10195 on display 0
10-30 20:54:59.291 D/KP2AAF  (4650): event: 2048, package = eu.siacs.conversations
10-30 20:54:59.411 I/ActivityManager(926): Displayed eu.siacs.conversations/.ui.ManageAccountActivity: +124ms
10-30 20:54:59.465 D/KP2AAF  (4650): event: 2048, package = eu.siacs.conversations
10-30 20:54:59.606 D/KP2AAF  (4650): event: 32, package = eu.siacs.conversations
10-30 20:54:59.706 I/cm.log.servpro(25167): [forcestop_listener][D]/  onAccessibilityEvent eu.siacs.conversations.ui.ManageAccountActivity, mListened = false
10-30 20:54:59.718 D/KP2AAF  (4650): event: 2048, package = eu.siacs.conversations
10-30 20:55:00.496 D/KP2AAF  (4650): event: 2048, package = eu.siacs.conversations
10-30 20:55:01.082 D/KP2AAF  (4650): event: 32, package = eu.siacs.conversations
10-30 20:55:01.165 D/KP2AAF  (4650): event: 2048, package = eu.siacs.conversations
10-30 20:55:02.356 D/conversations(17722): roi@jabber.hot-chilli.net: signing status message ""
10-30 20:55:02.374 D/KP2AAF  (4650): event: 2048, package = eu.siacs.conversations
10-30 20:55:02.478 D/KP2AAF  (4650): event: 32, package = eu.siacs.conversations
10-30 20:55:02.575 I/cm.log.servpro(25167): [forcestop_listener][D]/  onAccessibilityEvent eu.siacs.conversations.ui.ManageAccountActivity, mListened = false
10-30 20:55:06.056 D/conversations(17722): app switched into background
```

**Comment by msebald** [comment]
I digged into this again. Conversations is publishing the wrong PGP key. I contact of me sees this in his mcabber client:

```
12-12 16:16 *** jid:  <xxxxx@jabber.hot-chilli.net>                                                                                           
12-12 16:16 *** Name: xxxxx@jabber.hot-chilli.net                                                                                             
12-12 16:16 *** Type: user                                                                                                                  
12-12 16:16 *** Subscription: both                                                                                                          
12-12 16:16 *** Resource: [o] (0) Workstation                                                                                               
                Status timestamp: 2016-12-12 16:05:32                                                                                       
                PGP key id: 1765FBA09CE31CC9                                                                                                
                Last PGP signature: good                                                                                                    
12-12 16:16 *** Resource: [o] (0) phone                                                                                                     
                Status timestamp: 2016-12-12 16:15:33                                                                                       
                PGP key id: yyyyy
```

xxxxx is the user part of my JID and yyyyy is the PGP key id.

I cannot say why Conversations is using the wrong PGP key. There is a key on the private keyring which matches exactly my JID. So this should be used. But it isn't, a contact can see this and I as a user on the device can see this, too, as the wrong password and key owner name is asked/displayed.

Please take a look into this. Thank you.

**Comment by msebald** [comment]
Update: I deleted the "wrong" private key in Openkeychain and did publish PGP key in Conversations again. And Hello, there is a box where I was asked about the key I want to use. I chose the correct key and restored my backup of Openkeychain to have the other key in again as well.

So in my eyes there is an option missing here in Conversations. Once a key is selected it is not possible to change this again.
